File: test/src/main/java/top/bettercode/summer/test/BaseTest.java
package top.bettercode.summer.test;

import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.TestPropertySource;
import org.springframework.util.StringUtils;
import top.bettercode.summer.web.support.ApplicationContextHolder;

/**
 * 基础测试类
 *
 * @author Peter Wu
 */
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
@TestPropertySource(properties = {
    "summer.security.enabled=false",
    "seata.enabled=false",
})
public abstract class BaseTest extends BaseLogTest {


  protected boolean embeddedDatabase() {
    return !StringUtils.hasText(ApplicationContextHolder.getProperty("spring.datasource.url"));
  }
}

File: test/src/main/kotlin/top/bettercode/summer/test/BaseTest.kt
package top.bettercode.summer.test

import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.TestPropertySource
import top.bettercode.summer.web.BaseController
import top.bettercode.summer.web.support.ApplicationContextHolder
import java.util.function.Supplier

/**
 * 基础测试类
 *
 * @author Peter Wu
 */
@Suppress("SpringBootApplicationProperties")
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
@TestPropertySource(properties = ["summer.security.enabled=false", "seata.enabled=false"])
abstract class BaseTest : BaseLogTest() {
    protected fun embeddedDatabase(): Boolean {
        return ApplicationContextHolder.getProperty("spring.datasource.url").isNullOrBlank()
    }

    @JvmOverloads
    protected fun unauth(msg: String? = null): Supplier<out RuntimeException?> {
        return BaseController.unauth(msg)
    }

    @JvmOverloads
    protected fun notFound(msg: String? = null): Supplier<out RuntimeException?> {
        return BaseController.notFound(msg)
    }

}
